src/app/games/connect-six/ConnectSixMove.ts

```diff
@@ -11,23 +11,22 @@ export class ConnectSixFirstMove extends MoveCoord {
     public static isInRange(coord: Coord): boolean {
         return coord.isInRange(ConnectSixState.WIDTH, ConnectSixState.WIDTH);
     }
-    public static from(coord: Coord): ConnectSixFirstMove {
+    public static from(coord: Coord): MGPFallible<ConnectSixFirstMove> {
```

Comment:
qui de faire des .of quand on sait qu'on va toujours réussir à construire le mouvement ? Car chaque .get() dans le code est potentiellement une erreur générée au runtime, donc généraliser les `from` c'est polluer le code avec des `.get()`

Reply:
si on sais que ça marche à tout les coup, en quoi le .get() est une erreur potentielle ?
En revanche J'AVOUE p-e que c'est moche que qu'ils méritent bien d'ajouter du .fromOf et .fromFrom dans le generator.

Tiens j'y pense, si on passe LaClasseEntière en param, le code pourrait lui même savoir si y a du .from ou du .of, tu penserais quoi de cette approche ?

Reply:
si on sait que ça marche à tous les coup, on sait pas si demain on va pas faire un changement qui fait péter un `.get()` car ça marche plus à tous les coups. Si on fait ce changement en gardant la distinction .of/.from, on devra obligatoirement repasser sur tous les .of pour les changer en .from et bien gérer les cas d'erreur. Donc c'plus safe !

Si on passe la classe entière faudrait faire du `['from']`, hmm pourquoi pas, sauf si on a des cas irrégulier (fromCoords ou autre)

Reply:
Bon là ayant reverté of/from, on est bon ?